### ISSUE TITLE
KC4.1.3: Globus Connect Endpoint Setup

### ISSUE THREAD

**Issue by KCB13** [heliumdatacommons/MVP-repo#7]
Owner: Claris 
Doer(s): @cmball1

**Comment by cmball1** [comment]
@clarisca 

It seems there are 2 types of [endpoints](https://docs.globus.org/faq/globus-connect-endpoints/).
1. Globus Connect Personal - used to create an endpoint on a single-user system, e.g. on a laptop or a personal desktop machine.
2. Globus Connect Server - used to create an endpoint on a multi-user system, e.g. on a campus HPC cluster or a lab server.

- [ ] I can set up a VM in Azure to use as an endpoint if it is easier or RENCI can provide the VM, whichever is easier. 
- [ ] Do we need both types of endpoints, or does the endpoint type not matter from the iRODS perspective?
- [ ] Will we need to connect to an endpoint on AWS S3 or Google Drive for testing? If so, we will need to workout a subscription and get these "premium" storage connectors.

**Comment by clarisca** [comment]
We will need both endpoints.  However, my recommendation would be to start first with Globus Connect Personal "desktop machine".  @jasoncoposky ?

I have submitted a request for a VM at RENCI. It should be up EOD today.

We do have subscriptions in Azure and AWS.  Let me ask, why do we need to connect to an endpoint on AWS S3 or Google Drive for testing this feature?

**Comment by cmball1** [comment]
@clarisca It seems as though iRODS might use the [Globus API](https://docs.globus.org/api/transfer/), in which case the Endpoint source would likely not matter. If the source does matter, we may want to test those cases too. 

I just wanted to ask to confirm either way.

**Comment by clarisca** [comment]
@cmball1:  could you please create an account to access the VM at RENCI?
https://access.renci.org/

**Comment by cmball1** [comment]
@clarisca I have requested an account at the provided link.

As far as I know a small VM with 100 GB of storage should be fine. We can drop a bunch of files into the storage later for testing.

**Comment by clarisca** [comment]
Centos 7?

**Comment by cmball1** [comment]
Yes, that should be fine.

**Comment by clarisca** [comment]
@cmball1 : 'globus.renci.org' (152.54.3.39) is done, and /projects/datacommons/globus created. You should be able to login with your rti account.

**Comment by cmball1** [comment]
@clarisca I'm not sure I understand how to connect to the VM. Should I be able to SSH into it? 
For example: `ssh <username>@152.54.3.39`?

**Comment by cmball1** [comment]
@clarisca 

1. I am able to SSH into the RENCI VM as described by Robert Jones from the RENCI helpdesk. I am not able to "Activate" the VM as an endpoint because I can not log into the VM from the Globus UI. I was able to install all the packages etc.. I have replied to the email to the RENCI helpdesk to find out if there is a way to do this. 

2. Once I create an endpoint, I cannot add another user to the endpoint unless we have a subscription. I can make the endpoint "public" where anyone with a Globus login could find and transfer files out of the endpoint.

**Comment by clarisca** [comment]
@cmball1 : I am working on (1). Regarding (2) I will reach out to Ian Foster to see if we can get a temporary subscription. Stay tuned.

**Comment by jasoncoposky** [comment]
I have managed to log into globus with my unc credentials.  Would that be sufficient for a test?

**Comment by clarisca** [comment]
This would be ideal.I am not too excited about asking for a "free temp subscription".

**Comment by cmball1** [comment]
@jasoncoposky Does UNC have a Globus Plus subscription? You can see what your account has access to, once logged in -> Account in top right, then check under the "Globus Plus". If so, I can give you all the information to set up the endpoint once Claris gets the other side from IT figured out.

@clarisca It looks like anyone can get a 90 Free trial [here](https://www.globus.org/subscriptions/non-profit-subscription-inquiry). We would't necessarily need to ask Ian Foster directly for a temporary subscription. (Not sure which way is best if UNC does not have a subscription.)

**Comment by jasoncoposky** [comment]
It does not appear that we have a globus plus subscription through UNC.

**Comment by cmball1** [comment]
@clarisca @jasoncoposky I have contacted Globus through the provided webform to obtain a 90 day free trial. I will keep you updated with the status of this.

**Comment by cmball1** [comment]
@jasoncoposky what's your UNC ID that you signed up with in Globus?

@clarisca I have the Group setup in Globus with the free trial subscription. I can add anyone to it that needs access. 

I'm still waiting to hear back from RENCI ITS to get the endpoint functioning.

**Comment by cmball1** [comment]
@clarisca With RENCI IT's help, the RENCI VM is set up as a managed endpoint in Globus. 

1. I was able to transfer a file from my personal computer to the RENCI VM using Globus.
2. I have a group set up in Globus for the Plus subscription. I can add anyone's Globus ID to the group.
3. The credentials in Globus for the RENCI VM need to be "reactivated" every 24 hours. This might be a restriction on credentials from RENCI. I'm not sure if there is anything we can do about this.
4. I emailed back to RENCI IT, I don't have access to create files in `/projects/datacommons/globus/` on the VM.

**Comment by jasoncoposky** [comment]
@cmball1 Can you add jcoposky@unc.edu to your group?

**Comment by cmball1** [comment]
@jasoncoposky You should receive an email invite shortly. Let me know if there is a problem. I can elevate your group permissions after you accept the invite.

**Comment by rayi113** [comment]
Similarly, @cmball1 could you add rayi@unc.edu also?  Thanks.

**Comment by clarisca** [comment]
Please add claris@unc.edu as well.

**Comment by cmball1** [comment]
@jasoncoposky @rayi113 You both now have Administrator privileges for the group. Let me know if you need anything else.

**Comment by cmball1** [comment]
@clarisca Invite sent.

**Comment by cmball1** [comment]
I have used the managed endpoint to create a [share](https://docs.globus.org/globus-connect-server-installation-guide/#sharing_section) in Globus. This appears as an endpoint that shares a specific directory on the RENCI VM. The credentials do not expire for this.

I have increased the credentials to expire every 7 days through [this](https://docs.globus.org/faq/transfer-sharing/#how_can_i_activate_an_endpoint_for_the_maximum_amount_of_time) information.

The hour limit on authorization seems to be imposed from the Globus MyProxy authentication mechanism. Information from Globus support:

```
The best you can do is to request a credential from the MyProxy server with a lifetime equal to the maximum credential lifetime it will issue.
By default, endpoints using MyProxy based activation can issues credentials that are good for up to 7 days - control with the '--myproxy-lifetime' option for the 'globus endpoint-activate' command. Endpoints using MyProxy-OAuth based activation issue credentials that are good for 7 days by default. Endpoints using CILogon based activations have activations that are good for 10 days by default. In the web interface, you can change the requested lifetime for your MyProxy credential when activating on an endpoint by clicking the 'advanced' link and specifying a new value (in hours) for the Credential Lifetime box. You can specify any value you like, but the MyProxy server will never issue a credential with a duration longer than the server's configured maximum credential lifetime. If you are activating on an endpoint via the API, then there is an analogous method to specify the requested duration for the lifetime of the issued MyProxy credential.
See:
https://docs.globus.org/faq/transfer-sharing/#how_can_i_activate_an_endpoint_for_the_maximum_amount_of_time

Another alternative is if you do not wish to be tethered to having to renew your activations, then you should look into creating Shared Endpoints for the data sources and destinations you wish to use. Access to a Shared Endpoint is controlled via ACLs set for a particular Globus identity, and thus doesn't rely on endpoint activation as traditional endpoint access does. One catch.... Shared Endpoints can only be hosted on Managed endpoints, such as those endpoints covered under a subscription plan, which RTI currently has. 

You can read about sharing below: 
Sharing/Shared Endpoints: 
https://www.globus.org/data-sharing 
and 
https://docs.globus.org/resource-provider-guide/#sharing_section

```

**Comment by KCB13** [comment]
This issue is now closed, please refer to #22 for updates
